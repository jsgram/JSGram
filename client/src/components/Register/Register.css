.hyperlink{
  color:red;
}
.cat{
  width:100%;
  height:100%;
}
/* .silver{
  position:absolute;
  height:30%;
  width:70%;
  margin-top:-5px;
  margin-left:0px;
} */
.row-bottom{
  height:50px;
}