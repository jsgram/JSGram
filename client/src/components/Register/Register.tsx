import React from 'react';
import { Field } from 'redux-form';
import { Link } from 'react-router-dom';
import { Button, Form, FormGroup, FormProps, Spinner } from 'reactstrap';
import { renderField } from '../commonComponents/reduxFormFields';
import '../styles/style.scss';
import logo from '../assets/logo.png';

class Register extends React.Component<any> {
    public render(): JSX.Element {
        const {handleSubmit, onSubmit, submitting}: FormProps = this.props;
        return (
            <div>
                <Form className='mt-4 bg-white' onSubmit={handleSubmit(onSubmit)}>
                    <div className='border'>
                        <FormGroup className='col-lg-10 offset-lg-1 text-center'>
                            <Link to='/'><img className='picture' src={logo} alt='logo'/></Link>
                            <p>Sign up to see photos from your friends.</p>
                            <p>
                                <i className='fa fa-google-plus'/>
                                <a href={process.env.REACT_APP_GOOGLE_AUTH_URL}
                                   className='text-danger login_google pl-2'>Log in with Google</a>
                            </p>
                            <div className='or-devider'>
                                <span></span>OR<span></span>
                            </div>
                            <Field
                                name='username'
                                type='text'
                                component={renderField}
                                label='username'
                                placeholder='Username'
                                className='form-control form-control-lg mt-3'
                            />
                            <Field
                                name='fullName'
                                type='text'
                                component={renderField}
                                label='fullname'
                                placeholder='Fullname'
                                className='form-control form-control-lg mt-3'
                            />
                            <Field
                                name='email'
                                type='text'
                                component={renderField}
                                label='email'
                                placeholder='E-mail'
                                className='form-control form-control-lg mt-3'
                            />
                            <Field
                                name='password'
                                type='password'
                                component={renderField}
                                label='Password'
                                placeholder='Password'
                                className='form-control form-control-lg mt-3'
                            />
                            <Button
                                className='mt-3'
                                disabled={submitting}
                                color='danger'
                                size='lg'
                                block
                            >
                                Register
                            </Button>
                            {submitting && <Spinner className='mt-3' color='dark'/>}
                        </FormGroup>
                        <div className='text-center col-lg-10 offset-lg-1'>
                            <p>
                                By signing up, you agree to share high quality photos of cats on
                                a daily basis.
                            </p>
                        </div>
                    </div>
                </Form>
                <Form className='bg-white mt-3'>
                    <div className='border'>
                        <FormGroup className='text-center register_acc mt-2'>
                            <p className='pt-2'>
                                Have an account?
                                <Link to='/login' className='pl-1 text-danger'>
                                    Log in
                                </Link>
                            </p>
                        </FormGroup>
                    </div>
                </Form>
            </div>
        );
    }
}

export default Register;
