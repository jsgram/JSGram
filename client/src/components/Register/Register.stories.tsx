import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../App';
import Register from '../../containers/RegisterContainer';

storiesOf('Register', module)
    .addDecorator((story: any): any => <Provider store={store}>{story()}</Provider>)
    .addDecorator((story: any): any => <MemoryRouter>{story()}</MemoryRouter>)
    .add('register', () => <Register/>);
