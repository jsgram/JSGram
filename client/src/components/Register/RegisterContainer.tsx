import React from 'react';
import Register from './Register';
import { reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { registerUser } from '../../store/register/actions';
import validate from '../../utils/validation';
import { FormProps } from 'reactstrap';
import { IUser } from '../../store/commonInterfaces/commonInterfaces';

class RegisterContainer extends React.Component<any> {
    constructor(props: any) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
    }

    public onSubmit(user: IUser): IUser {
        return this.props.registerUser(user);
    }

    public render(): JSX.Element {
        const {handleSubmit, submitting}: FormProps = this.props;
        return (
            <Register
                handleSubmit={handleSubmit}
                onSubmit={this.onSubmit}
                submitting={submitting}
            />
        );
    }
}

export default connect(
    null,
    {registerUser},
)(
    reduxForm({
        form: 'registerForm',
        validate,
    })(RegisterContainer),
);
