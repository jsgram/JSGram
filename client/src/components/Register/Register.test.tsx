import Register from './index';
import * as reactstrap from 'reactstrap';

import { shallow } from 'enzyme';
import React from 'react';

describe('Register component:', () => {
    let renderer;

    beforeEach(() => {
        reactstrap.Button = jest.fn(() => (<div></div>));
        reactstrap.Form = jest.fn(() => (<div></div>));
        reactstrap.FormGroup = jest.fn(() => (<div></div>));
        reactstrap.Spinner = jest.fn(() => (<div></div>));

        const handleSubmit = jest.fn(() => 'somehandlesubmit');
        renderer = shallow(<Register handleSubmit={handleSubmit} />);
    });

    test.skip('render - success', () => {
        expect(renderer.html()).toHaveLength(33);
    });
});
