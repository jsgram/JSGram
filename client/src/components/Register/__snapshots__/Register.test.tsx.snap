// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component: render - success 1`] = `ShallowWrapper {}`;
